### DISCUSSION TITLE
如何关掉回调

### DISCUSSION THREAD

**Discussion by sichal** [wondertrader/wondertrader#44] [Q&A] [answered]
from wtpy import WtEngine,EngineType
#from Strategies.DualThrust import StraDualThrust
from Strategies.GtridTrade import GridStra
from ConsoleIdxWriter import ConsoleIdxWriter

if __name__ == "__main__":
    #创建一个运行环境，并加入策略
    print("befor wtengin")
    env = WtEngine(EngineType.ET_CTA)
    print("after wtengin")
    env.init('../common/', "config.yaml")
    print("CTA引擎:",env)
    #straInfo = StraDualThrust(name='pydt_rb', code="SHFE.rb.HOT", barCnt=50, period="m5", days=30, k1=0.2, k2=0.2, isForStk=False)
    straInfo = GridStra(name='gridstra_fg', code="CZCE.FG.2209", barCnt=50, period="m1", short_days=3, long_days=15, num=5, p1=0.01, p2=-0.005,capital=500000)
    env.add_cta_strategy(straInfo)
    
    idxWriter = ConsoleIdxWriter()
    env.set_writer(idxWriter)

    env.run()

    kw = input('press any key to exit\n')

上面是demo代码，当我想中途停止策略时，按键后回车，回调还在继续，接口里没有找到关闭回调的函数，也没看到关闭服务的接口

**Comment by wondertrader** [comment]
Control+C不能够中断吗？

**Comment by wondertrader** [comment, marked as answer]
有一个方案，你最后启动调用run的时候，可以尝试以下方式：
`engine.run(bAsync=True)`

**Comment by sichal** [comment]
经过作者提示，直接在命令窗口启动，control+c是能结束掉的，在sypder的console里是无法结束的，所以想看临时print的内容的话，不要在spyder里跑
